src/engine-actions/prepare-swap.js

```diff
@@ -34,19 +34,12 @@ async function prepareSwap (orderId, swapHash, value) {
     memo: `${MEMO_PREFIX}${orderId}`,
     value,
     expiry: SWAP_EXPIRY,
-    externalPreimage: true,
-    rHash: swapHash
+    hash: swapHash
   }
 
-  const { rHash } = await addInvoice(params, { client: this.client })
+  const { invoice } = await addHoldInvoice(params, { client: this.client })
 
-  if (rHash !== swapHash) {
-    throw new Error(
-      `Error while preparing for swap: returned rHash did not match passed swapHash: rHash=${rHash}, swapHash=${swapHash}`
-    )
-  }
-
-  return swapHash
+  return invoice
```

Comment:
Why change the return value here? Why not keep it as the hash?

Reply:
My thinking was that the caller already has `swapHash` since it was passed in, and the invoice is what is being generated in this method and will be lost if we don't return it (or would have to be looked up some other way). Since the call site isn't using the return value it looks like we can change this without affecting anything else. I need to fix this either way though because it's supposed to be called `paymentRequest`.

Reply:
Ok that makes sense.